crates/bevy_math/src/smooth_damp.rs

```diff
@@ -0,0 +1,170 @@
+    ///     let SmoothDamp { position, velocity } = Vec3::smooth_damp(
+    ///         transform.translation,
+    ///         smoother.target,
+    ///         smoother.velocity,
+    ///         smoother.smoothness,
+    ///         dt,
+    ///     );
+    ///     transform.translation = position;
+    ///     smoother.velocity = velocity;
+    /// }
+    /// ```
+    fn smooth_damp(
+        from: Self,
+        to: Self,
+        velocity: Self,
+        smooth_time: f32,
+        delta_time: f32,
+    ) -> SmoothDamp<Self>
+    where
+        Self: Sized;
+
+    /// Smooths value to a goal using a damped spring limited by a maximum speed.
+    ///
+    /// `smooth_time` is the expected time to reach the target when at maximum velocity.
+    ///
+    /// Returns smoothed value and new velocity.
+    ///
+    /// # Example
+    /// ```
+    /// # use bevy_math::prelude::{Vec3, Quat};
+    /// # use bevy_math::{SmoothDamp, SmoothDampFunctions};
+    /// # struct Transform {
+    /// #     translation: Vec3,
+    /// #     rotation: Quat,
+    /// #     scale: Vec3
+    /// # }
+    /// struct SmoothTransform {
+    ///     pub smoothness: f32,
+    ///     pub max_speed: f32,
+    ///     pub target: Vec3,   
+    ///     velocity: Vec3   
+    /// }
+    ///
+    /// fn smooth_transform_update(dt: f32, transform: &mut Transform, smoother: &mut SmoothTransform) {
+    ///     let SmoothDamp { position, velocity } = Vec3::smooth_damp_max(
+    ///         transform.translation,
+    ///         smoother.target,
+    ///         smoother.velocity,
+    ///         smoother.max_speed,
+    ///         smoother.smoothness,
+    ///         dt,
+    ///     );
+    ///     transform.translation = position;
+    ///     smoother.velocity = velocity;
+    /// }
+    /// ```
+    #[track_caller]
+    fn smooth_damp_max(
+        from: Self,
+        to: Self,
+        velocity: Self,
+        max_speed: f32,
+        smooth_time: f32,
+        delta_time: f32,
+    ) -> SmoothDamp<Self>
+    where
+        Self: Sized;
+}
+
+macro_rules! impl_smooth_damp {
+    ($t:ty, $f:ty, $clamp:expr) => {
+        impl SmoothDampFunctions for $t {
+            #[inline]
+            fn smooth_damp(
+                from: $t,
+                to: $t,
+                velocity: $t,
+                smooth_time: f32,
+                delta_time: f32,
+            ) -> SmoothDamp<Self> {
+                let smooth_time = <$f>::max(smooth_time as $f, 0.00001);
+                let delta_time = delta_time as $f;
+
+                // from game programming gems 4, chapter 1.10
+                let omega = 2.0 / smooth_time;
+                let x = omega * delta_time;
+
+                // fast and good enough approximation of exp(x)
+                let exp = 1.0 / (1.0 + x * (1.0 + x * (0.48 + 0.235 * x)));
+
+                let change = from - to;
+                let temp = (velocity + omega * change) * delta_time;
+
+                SmoothDamp::<$t> {
```

Comment:
```suggestion
                SmoothDamp {
```
I have a feeling the `::<$t>` may be unnecessary, though I might be wrong...